### PR TITLE
Feature swig interface 

### LINKED ISSUE #261: SWIG integration for multilanguage interface
Expose the interfaces following classes:

- [x] ReturnData
- [x] Solver
- [x] ExpData
- [x] Model
- [x] amici.dpp
- [x] simulation routine

will make compiled AMICI functions usable from R, Python, JS, etc.

### PR DESCRIPTION
**Pull request by FFroehlich**
fixes #261 

currently only implements python interface but other languages could be added with a few lines of code

### CODE REVIEW

**Comment by FFroehlich** [comment]
@dweindl I currently receive the following error when building:
```
/Users/F.Froehlich/Documents/MATLAB/AMICI/models/model_dirac/build/model_diracPYTHON_wrap.cxx:3281:39: error: call to implicitly-deleted copy constructor of
      'std::unique_ptr<amici::Model>'
  resultobj = SWIG_NewPointerObj((new std::unique_ptr< amici::Model >(static_cast< const std::unique_ptr< amici::Model >& >(result))), SWIGTYPE_p_std__unique_ptrT_amici__Mod...
```
similar problems appear when removing `%ignore getSolver` from `model.i`/`model_ode.i`/`model_dae.i`

File: CMakeLists.txt [hunk 1]
```diff
@@ -100,7 +102,11 @@ if($ENV{ENABLE_GCOV_COVERAGE})
 	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g -O0 --coverage")
 endif()
 
-include(CMakeModules/clang-tools.cmake)
+include(clang-tools)
+
+set(AUTHORS "Fabian Froehlich, Jan Hasenauer, Daniel Weindl and Paul Stapor")
+set(AUTHOR_EMAIL "fabian.froehlich@helmholtz-muenchen.de")
+set(PROJECT_VERSION "0.0.0a0")
```
Comment (dweindl) [inline CMakeLists.txt]: TODO: Set proper version

File: python/amici/sbml_import.py [hunk 2]
```diff
@@ -0,0 +1,936 @@
...
+        sbml_doc: private?
+        sbml: private?
+        modelname:
+        modelPath:
+        self.modelSwigPath:
+        self.functionBodies:
+    """
+
+    # TODO: are units respected on sbml import? if not convert; at least throw if differ?
+    # TODO: camelCase?
```
Comment (dweindl) [inline python/amici/sbml_import.py]: TODO: cleanup and document attributes

File: python/amici/setup.template.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    """Get list of source files for the amici base library"""
+    import glob
+    import re
+    modelSources = glob.glob('*.cpp')
+    return modelSources
+
+
+def getAmiciLibs():
+    """Get list of libraries for the amici base library"""
+    ''' TODO: scrambles order
```
Comment (dweindl) [inline python/amici/setup.template.py]: TODO cleanup

File: python/setup.py.in [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+def getAmiciBaseSources():
+    """Get list of source files for the amici base library"""
+    import glob
+    import re
+    amiciBaseSources = glob.glob('amici/src/*.cpp')
+    amiciBaseSources = [src for src in amiciBaseSources if not re.search(r'(matlab)|(\.template\.)', src)]
+    return amiciBaseSources
+
+"""TODO
```
Comment (dweindl) [inline python/setup.py.in]: TODO: cleanup

File: python/setup.py.in [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+                             ],
+                         library_dirs=[
+                             *h5pkgcfg['library_dirs'],
+                             'amici/libs/',
+                                       ],
+                         swig_opts=['-c++', '-modern', '-v', '-Iamici/include/', '-outdir', 'amici/'],
+                         extra_link_args=['${BLAS_LIBRARIES}'],
+                         extra_compile_args=cxx_flags
+                         )
+"""
```
Comment (dweindl) [inline python/setup.py.in]: TODO cleanup

File: python/test/testSBML.py [hunk 6]
```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+
+import traceback
+import os
+import sys
+import importlib
+import numpy as np
+import amici
+
+# directory with sbml semantic test cases
```
Comment (dweindl) [inline python/test/testSBML.py]: TODO document

File: tests/testModels.py [hunk 7]
```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python3
```
Comment (dweindl) [inline tests/testModels.py]: TODO: document

File: CMakeLists.txt (hunk 1 above)
Comment (dweindl) [reply CMakeLists.txt]: done

File: tests/testModels.py (hunk 7 above)
Comment (FFroehlich) [reply tests/testModels.py]: done

**Pull request merged**